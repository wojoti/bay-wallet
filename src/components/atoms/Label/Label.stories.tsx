import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import Label from './Label';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Atoms/Label',
  component: Label,
  argTypes: {
    color: {
      control: {type: 'color'},
    },
    align: {
      control: {type: 'select'},
      options: ['left', 'auto', 'right', 'center', 'justify', undefined],
    },
  },
} as ComponentMeta<typeof Label>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Label> = args => <Label {...args} />;

export const Example = Template.bind({});
Example.args = {
  children: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  color: 'black',
  size: 14,
  testId: '',
  align: 'left',
};

export const CenteredExample = Template.bind({});
CenteredExample.args = {
  children: 'Lorem ipsum dolor sit amet, consectetur adipiscing',
  color: '#6B6B6B',
  size: 20,
  testId: '',
  align: 'center',
};
