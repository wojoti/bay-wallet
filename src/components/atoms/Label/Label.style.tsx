import {StyleSheet} from 'react-native';
import {defaultWeight, font} from 'theme/fonts';

interface Props {
  color?: string;
  size?: number;
  align?: 'left' | 'auto' | 'right' | 'center' | 'justify';
}
const style = ({color, size, align}: Props) =>
  StyleSheet.create({
    wrapper: {
      color: color || 'black',
      textAlign: align || 'left',
      fontSize: size || 14,
      fontFamily: font,
      fontWeight: defaultWeight,
    },
  });
export default style;
