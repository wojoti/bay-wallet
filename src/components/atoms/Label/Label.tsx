import {PropsWithChildren} from 'react';
import {Text} from 'react-native';
import style from './Label.style';

export type LabelProps = PropsWithChildren<{
  color?: string;
  testId?: string;
  align?: 'left' | 'auto' | 'right' | 'center' | 'justify';
  size?: number;
}>;

const Label = ({color, align, size, testId, children}: LabelProps) => {
  return (
    <Text style={style({color, size, align}).wrapper} testID={testId}>
      {children}
    </Text>
  );
};

export default Label;
