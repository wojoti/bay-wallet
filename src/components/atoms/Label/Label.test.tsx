import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Label, {LabelProps} from './Label';

const props: LabelProps = {
  testId: 'test-label-id',
};

const modifiedProps: LabelProps = {
  ...props,
  color: 'red',
  size: 100,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(
    <Label {...props}>test label</Label>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render label', () => {
  render(<Label {...props}>test label</Label>);
  const labelElement = screen.getByTestId('test-label-id');
  expect(labelElement).toBeOnTheScreen();
});

test('should render label with text', () => {
  render(<Label {...props}>test label</Label>);
  const labelElement = screen.getByTestId('test-label-id');
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveTextContent('test label');
});

test('should render label with modified style through props', () => {
  render(<Label {...modifiedProps} />);
  const labelElement = screen.getByTestId('test-label-id');
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveStyle({
    color: 'red',
    fontSize: 100,
  });
});
