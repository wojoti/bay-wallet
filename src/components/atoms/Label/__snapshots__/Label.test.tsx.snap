// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Text
  style={
    {
      "color": "black",
      "fontFamily": "Poppins-Light",
      "fontSize": 14,
      "fontWeight": "400",
      "textAlign": "left",
    }
  }
  testID="test-label-id"
>
  test label
</Text>
`;
